File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int n = heights.size(), vol = 0;
        for(int i=0;i<n-1;i++){
            for (int j=i+1;j<n;j++){
                int mini = min(heights[i], heights[j]);
                vol = max(mini*(j-i), vol);
            }
        }
        return vol;
    }
};

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        n = len(heights)
        area = 0
        for i in range (n):
            mini = 1001
            for j in range (i, n):
                width = j-i+1
                mini = min(mini, heights[j])
                area = max(area, width*mini)
            
        return area
            